.github/workflows/cd.yml

```diff
@@ -37,24 +37,54 @@ jobs:
             echo "SERVICE_NAME=${{ vars.AWS_ECS_API_SERVICE_STAGING }}" >> $GITHUB_ENV
           fi
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
+      - name: Check for existing image with GIT_SHA
+        env:
+          GIT_SHA: ${{ github.sha }}
+          REPO: ${{ vars.AWS_ECR_API_REPO }}
+        run: |
+          image_manifest=$(aws ecr batch-get-image --repository-name $REPO --image-ids imageTag=$GIT_SHA --query 'images[0].imageManifest')
+          echo "IMAGE_MANIFEST=$image_manifest" >> $GITHUB_ENV
+
+      - if: env.IMAGE_MANIFEST == 'null'
```

Comment:
why would IMAGE_MANIFEST be the string value "null" ?

Reply:
Correct. Since we're storing the result in an environment variable it gets cast to a string which is why we need to check for the `'null'` string value.